File: web/default/src/lib/request-health.ts
export type RequestHealthStatus = 'healthy' | 'unstable' | 'failed' | 'unknown'

const REQUEST_HEALTH_LABELS: Record<RequestHealthStatus, string> = {
  healthy: '稳定',
  unstable: '波动',
  failed: '异常',
  unknown: '暂无近期请求',
}

export function classifyRequestHealth(
  successRate: number | null | undefined,
  requestCount: number | null | undefined
): RequestHealthStatus {
  if (
    !requestCount ||
    requestCount <= 0 ||
    successRate == null ||
    !Number.isFinite(successRate)
  ) {
    return 'unknown'
  }
  if (successRate > 90) return 'healthy'
  if (successRate >= 75) return 'unstable'
  return 'failed'
}

export function getRequestHealthLabel(status: RequestHealthStatus) {
  return REQUEST_HEALTH_LABELS[status]
}

File: internal/gateway/domain/request_health.go
package domain

const (
	RequestHealthHealthy  = "healthy"
	RequestHealthUnstable = "unstable"
	RequestHealthFailed   = "failed"
	RequestHealthUnknown  = "unknown"
)

// ClassifyRequestHealth maps one request sample window to the shared health
// vocabulary used by all group-status surfaces.
func ClassifyRequestHealth(successRate float64, requestCount int64) string {
	if requestCount <= 0 {
		return RequestHealthUnknown
	}
	if successRate > 90 {
		return RequestHealthHealthy
	}
	if successRate >= 75 {
		return RequestHealthUnstable
	}
	return RequestHealthFailed
}
